### PR TITLE
signal: Remove depecrated signal constant SIGUNUSED

### LINKED ISSUE #68: Compiling shim on ClearLinux gives error
Compiling shim on clearlinux version 17650, gives the following error```
```
src/utils.c:67:2: error: ‘SIGUNUSED’ undeclared here (not in a function); did you mean ‘SI_USER’?
  SIGUNUSED,
  ^~~~~~~~~
```
Turns out that this constant has been removed in glib2.6 that clearlinux 17650 uses.
Ref: https://sourceware.org/git/?p=glibc.git;a=blob;f=NEWS;h=dd381f89308d850e0513cae935fb02adb094845a;hb=HEAD#l198

### PR DESCRIPTION
**Pull request by amshinde**
With glibc 2.6, the obsolete signal constant SIGUNUSED is no longer
defined by signal.h.

Fixes #68

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Comment by gorozco1** [comment]
Is this only for distros with newer glibc? if so, what would happen with distros that still use older glibc?

File: src/utils.c [hunk 1]
```diff
@@ -64,7 +64,6 @@ int shim_signal_table[] = {
 	SIGIO,               /* I/O now possible */
 	SIGPWR,              /* Power failure restart */
 	//SIGSYS,            /* Bad system call */
-	SIGUNUSED,
```
Comment (jodh-intel) [inline src/utils.c]: How about `#ifdef`-ing this instead of simply removing?

Comment (grahamwhaley) [reply src/utils.c]: for if-deffing, +1 - we still want to work as we did for older versions of the lib/compiler I would think...

Comment (amshinde) [reply src/utils.c]: @jodh-intel I have added the #ifdef to include the signal constant if defined.

**Comment by jodh-intel** [comment]
lgtm

[Approved with PullApprove](https://pullapprove.com/clearcontainers/shim/pull-request/69/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/shim)

**Comment by gorozco1** [comment]
lgtm


[Approved with PullApprove](https://pullapprove.com/clearcontainers/shim/pull-request/69/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/shim)

**Review by grahamwhaley — APPROVED** [review]
thanks for the ifdef
lgtm

**Pull request merged**